### PR TITLE
monkey patch gnupg._parsers.DeleteResult._handle_status

### LINKED ISSUE #4294: GPG PINENTRY_LAUNCHED error when sources are deleted
## Description

Deleting a source raises this error.

```pytb
Exception in thread Thread-39:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pretty_bad_protocol/_meta.py", line 670, in _read_response
    result._handle_status(keyword, value)
  File "/usr/local/lib/python2.7/dist-packages/pretty_bad_protocol/_parsers.py", line 1024, in _handle_status
    raise ValueError("Unknown status message: %r" % key)
ValueError: Unknown status message: u'PINENTRY_LAUNCHED'
```

## Steps to Reproduce

- Start container
- Start client
- Delete source

## Expected Behavior

No errors.

## Actual Behavior

There are errors but the HTTP code is still 200, and the key is still deleted on the server side.

### PR DESCRIPTION
**Pull request by redshiftzero**
## Status

Ready for review

## Description of Changes

Fixes #4294.

Changes proposed in this pull request:
 - Monkey patch internal method to ensure that gpg's PINENTRY_LAUNCHED status does not raise a ValueError

## Testing

0. Delete a source
1. See no PINENTRY_LAUNCHED ValueError

## Deployment

app code only

### If you made changes to the server application code:

- [ ] Linting (`make ci-lint`) and tests (`make -C securedrop test`) pass in the development container

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### CODE REVIEW

File: securedrop/crypto_util.py [hunk 1]
```diff
@@ -37,6 +37,23 @@
                        'BCDEFGHIJKLMNOPQRSTUVWXYZ')
 
 
+def monkey_patch_delete_handle_status(self, key, value):
+        """Parse a status code from the attached GnuPG process.
+        :raises: :exc:`~exceptions.ValueError` if the status message is unknown.
+        """
+        if key in ("DELETE_PROBLEM", "KEY_CONSIDERED"):
```
Comment (emkll) [inline securedrop/crypto_util.py]: :+1: `DELETE_PROBLEM` should be the only other status when deleting the key (https://github.com/gpg/gnupg/blob/STABLE-BRANCH-2-2/g10/delkey.c#L80)

Where did you get `KEY_CONSIDERED`?

Comment (redshiftzero) [reply securedrop/crypto_util.py]: that's from upstream: so this method is https://github.com/isislovecruft/python-gnupg/blob/master/pretty_bad_protocol/_parsers.py#L1015 but with an additional `elif` block for the `PINENTRY_LAUNCHED` case

**Comment by emkll** [comment]
In Prod VMs (Trusty->Xenial install, which uses `pinentry-gtk2`), on app package built on this branch
- submit a document via source interface
- log into journalist interface and delete source
- [x] There are no errors in `/var/log/apache2/journalist-error.log`

note that per https://github.com/freedomofpress/securedrop/issues/4294#issuecomment-486378613 , this error does not seem to appear in instances upgraded from Trusty to Xenial.

**Comment by redshiftzero** [comment]
OK I've marked this ready to merge: we need this for instances with `pinentry-curses`

**Pull request merged**